keyboards/planck/ez/ez.c

```diff
@@ -112,39 +116,148 @@ void suspend_power_down_kb(void) {
 }
 #endif
 
-void matrix_init_kb(void) {
-  matrix_init_user();
+/* Left B9   Right B8 */
+
+// See http://jared.geek.nz/2013/feb/linear-led-pwm
+static uint16_t cie_lightness(uint16_t v) {
+  if (v <= 5243) // if below 8% of max
+    return v / 9; // same as dividing by 900%
+  else {
+    uint32_t y = (((uint32_t) v + 10486) << 8) / (10486 + 0xFFFFUL); // add 16% of max and compare
+    // to get a useful result with integer division, we shift left in the expression above
+    // and revert what we've done again after squaring.
+    y = y * y * y >> 8;
+    if (y > 0xFFFFUL) // prevent overflow
+      return 0xFFFFU;
+    else
+      return (uint16_t) y;
+  }
+}
+
+static PWMConfig pwmCFG = {
+    0xFFFF,/* PWM clock frequency  */
+    256,/* initial PWM period (in ticks) 1S (1/10kHz=0.1mS 0.1ms*10000 ticks=1S) */
+    NULL,
+    {
+        {PWM_OUTPUT_DISABLED, NULL}, /* channel 0 -> TIM1-CH1 = PA8 */
+        {PWM_OUTPUT_DISABLED, NULL}, /* channel 1 -> TIM1-CH2 = PA9 */
+        {PWM_OUTPUT_ACTIVE_HIGH, NULL},
+        {PWM_OUTPUT_ACTIVE_HIGH, NULL}
+    },
+    0, /* HW dependent part.*/
+    0
+};
 
-  palSetPadMode(GPIOB, 8, PAL_MODE_OUTPUT_PUSHPULL);
-  palSetPadMode(GPIOB, 9, PAL_MODE_OUTPUT_PUSHPULL);
+static uint32_t planck_ez_right_led_duty;
+static uint32_t planck_ez_left_led_duty;
 
-  palClearPad(GPIOB, 8);
-  palClearPad(GPIOB, 9);
+void planck_ez_right_led_level(uint8_t level) {
+    planck_ez_right_led_duty = (uint32_t)(cie_lightness(0xFFFF * (uint32_t) level / 255));
+    if (level == 0) {
+        // Turn backlight off
+        pwmDisableChannel(&PWMD4, 2);
+    } else {
+        // Turn backlight on
+        pwmEnableChannel(&PWMD4, 2, PWM_FRACTION_TO_WIDTH(&PWMD4,0xFFFF,planck_ez_right_led_duty));
+    }
 }
 
-void matrix_scan_kb(void) {
-  matrix_scan_user();
+
+void planck_ez_right_led_on(void){
+    pwmEnableChannel(&PWMD4, 2, PWM_FRACTION_TO_WIDTH(&PWMD4,0xFFFF,planck_ez_right_led_duty));
 }
 
-uint32_t layer_state_set_kb(uint32_t state) {
+void planck_ez_right_led_off(void){
+    pwmDisableChannel(&PWMD4, 2);
+}
+
+void planck_ez_left_led_level(uint8_t level) {
```

Comment:
I assume `planck_ez_right_led_level` and  `planck_ez_left_led_level` are not meant to be an externally consumed API? the whole `(uint8_t)keyboard_config.led_level * 255 / 4` is a bit unintuitive compared to the regular backlight API.

Reply:
Not really, but they can be.  

Mostly, wanted to get this working, and then later, make it a lot more user configurable.